api/v1/storagecluster_types.go

```diff
@@ -137,11 +140,41 @@ type ManageCephObjectStoreUsers struct {
 	ReconcileStrategy string `json:"reconcileStrategy,omitempty"`
 }
 
+// ExternalStorageKind specifies a kind of the external storage
+type ExternalStorageKind string
+
+const (
+	// KindOcs specifies a "ocs" kind of the external storage
+	KindOcs ExternalStorageKind = "ocs"
+
+	// KindRhcs specifies a "rhcs" kind of the external storage
+	KindRhcs ExternalStorageKind = "rhcs"
+)
+
 // ExternalStorageClusterSpec defines the spec of the external Storage Cluster
-// to be connected to the local cluster
+// to be connected to the external cluster
 type ExternalStorageClusterSpec struct {
-	// +optional
+	//+kubebuilder:default:=true
```

Comment:
Why `default: true`?

Reply:
Why do we need the enabled field at all, Why does the fact that we have this section on the CR enough for implying external storage?

Reply:
existing implementation using `enabled` field.

Reply:
Ok, so in that case, we need to keep it to be backward compatible. 
But what is the expected behaviour if someone changes the value from true to false after deployment?

Reply:
> Why default: true?

So that if someone adds any other field under the `externalStorage` in the CR this will get filled automatically to true.

> Why do we need the enabled field at all, Why does the fact that we have this section on the CR enough for implying external storage?

This is still an optional feild while create a CR taking below `omitempty` code into consideration. 

```
ExternalStorage ExternalStorageClusterSpec `json:"externalStorage,omitempty"`
```

Reply:
@iamniting  What about the question about the behavior upon changes to this field post deployment?

Reply:
Can you pls elaborate?

Reply:
Will setting the default to true mean that this will never be empty? I would prefer that it stay false to keep the behaviour the same.

Reply:
To clarify - what if someone creates an empty externalstorage struct in the StorageCluster. What is the behaviour with the default:true ?

Reply:
creating empty {} will be a problem as all these default values will be placed in the CR.

Reply:
does that mean `kubebuilder:default` and golang's `omitempty` are contradicting and should not be used together?

Reply:
In that case, please remove the default setting or set it to false.

Reply:
Setting it to false is problematic as it indicates that if it is not present we should not establish a connection to the remote server which is undefined behavior. If the sub-struct is present we should act upon it.

Reply:
Disagree - we use the enable value field to check how to setup the cluster and reconcile the values. Currently an empty ExternalStorage struct is treated as not External storage.
If the value is not present, it would be better to return an error until the struct is edited.

Reply:
But in that case what will happen if the value is reverted to false?

Reply:
I also don't believe there is a single-use case where the sub-struct is defined and the value is set to false. In an ideal world I will just stop watching that value in deprecate it (and in the future remove it completely)

Reply:
When and by who would it be reverted? We can call out that the value cannot be changed once set , the way we do for other fields.

Reply:
I have seen empty structs in yamls when I read them from the cluster - I want to ensure we do not run into such a problem say if a YAML is edited in the console. I am against using the default value in this case and I do not see a big requirement for setting the default.

Reply:
> When and by who would it be reverted? We can call out that the value cannot be changed once set
In that case, let's mark it as immutable. but then there is no point in having it in the first place as if the storage cluster is created with false in this field it is unusable

Reply:
This PR can either implement the new changes or fix an old implementation detail. We can't do both.
If you think what we're doing is wrong or doesn't make sense, please raise an issue with details and we can continue the discussion. For now, please revert this change @iamniting .

Reply:
@umangachapagain reverted the change.